src/Nebulas/Signer.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+#include "../HexCoding.h"
+
+using namespace TW;
+using namespace TW::Nebulas;
+
+const char *Signer::TxPayloadBinaryType = "binary";
+const char *Signer::TxPayloadDeployType = "deploy";
+const char *Signer::TxPayloadCallType = "call";
+
+void Signer::sign(const PrivateKey &privateKey, Transaction &transaction) const noexcept {
+    auto hash = this->hash(transaction);
+    transaction.algorithm = 1;
```

Comment:
can you use value from `SigningInput`? not hardcoded

Reply:
Ok, I will add a new function use the value by SigningInput.